### PR TITLE
view cohort page and students of cohorts

### PR DESCRIPTION
**Pull request by HemaSAli**
view cohort page and students of cohorts in admin panel
relates #5  #6

### CODE REVIEW

File: public/css/cohorts.css [hunk 1]
```diff
@@ -0,0 +1,166 @@
+/* section */
+.styled {
+  margin-right: 1rem;
+}
```
Comment (NouraldinS) [inline public/css/cohorts.css]: Add newline after curly brace.

File: public/css/cohorts.css [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+.button2 {
+  background-color: #389dbf;
+}
+
+.tabel {
+  border-collapse: collapse;
+  width: 100%;
+  margin-top: 2%;
+  margin-bottom:3%
+
```
Comment (NouraldinS) [inline public/css/cohorts.css]: Remove newline.

File: public/css/cohorts.css [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  font-weight: bold;
+}
+.button2 {
+  background-color: #389dbf;
+}
+
+.tabel {
+  border-collapse: collapse;
+  width: 100%;
+  margin-top: 2%;
```
Comment (NouraldinS) [inline public/css/cohorts.css]: Do not use relative margins unless you're willing to fight for it.

File: public/css/cohorts.css [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+.sectionAddcoh--visible{
+    display: block;
+    margin-bottom: 4rem;
+
+}
+/* sectionstd */
+
+.sectionstd--visible{
+    display: block;
+    margin-bottom: 7%
```
Comment (NouraldinS) [inline public/css/cohorts.css]: Do not use relative margins unless you're willing to fight for it.

File: public/css/cohorts.css [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+input[type="submit"] {
+  margin-top: 1rem;
+  background-color: #4caf50;
+  color: white;
+  padding: 0.8rem 2rem;
+  border: none;
+  border-radius: 4px;
+  cursor: pointer;
+}
```
Comment (NouraldinS) [inline public/css/cohorts.css:131]: Add newline after curly brace.

File: public/css/cohorts.css [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+input[type=submit]:hover {
+    background-color: #45a049;
+}
+
+.sectionAddstd{
+    display: none
+}
+.sectionAddstd--visible{
+    margin-top: 2%;
```
Comment (NouraldinS) [inline public/css/cohorts.css]: Fix indentation and don't use relative margins nor paddings.

File: public/js/cohort.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const addStd = document.getElementById('addStd');
+const addCoh = document.getElementById('addCoh');
+
+function addCohort() {
+  addCoh.classList.toggle('sectionAddcoh--visible');
+}
+
+// .then(result => console.log(result.header));
```
Comment (NouraldinS) [inline public/js/cohort.js]: Remove unnecessary comment.

File: public/js/cohortStudent.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
+const studentName = document.getElementById('studentName');
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: I'd prefer if you had a function that cleans your code a bit like:
```js
const getElementById = id => document.getElementById(id);
```

File: public/js/cohortStudent.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+const gitHubUserName = document.getElementById('gitHubUserName');
+const addStudentButton = document.getElementById('addStudentButton');
+const loading = document.getElementById('loading');
+const URL = window.location.href;
+const splitUrl = URL.split('/');
+const cohortID = splitUrl[splitUrl.length - 2];
+const deleteStudentButton = document.querySelectorAll('.delete');
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: Buttons*

File: public/js/cohortStudent.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+
+function addStudents() {
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: Either use functions or arrow functions, do not use both.

File: public/js/cohortStudent.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+  addStd.classList.toggle('sectionAddstd--visible');
+}
+
+addStudentButton.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const studentNameValue = studentName.value;
+  const gitHubUserNameValue = gitHubUserName.value;
+  if (
+    studentNameValue.trim().length !== 0 &&
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]: Empty strings already evaluate to false, so this:
```js
if(studentNameValue.trim())
```
should evaluate the same as
```js
if(studnetNameValue.trim().length === 0)
```

File: public/js/cohortStudent.js [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    const apiLink = `https://api.github.com/users/${gitHubUserNameValue}`;
+    fetch(apiLink, { method: 'GET' })
+      .then((loading.style.display = 'block'))
+      .then((response) => {
+        loading.style.display = 'none';
+        if (response.status !== 200) throw new Error('Invalid_username');
+        return response.json();
+      })
+      .then((response) => {
+        const { avatar_url } = response;
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]:
```js
const { avatar_url: avatarUrl, html_url: htmlUrl } = response;
```

File: public/js/cohortStudent.js [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+        };
+        fetch('/admin/newStudent', {
+          method: 'POST',
+          headers: { 'content-type': 'application/json' },
+          body: JSON.stringify(newStudent),
+        })
+          .then(result => result.json())
+          .then((result) => {
+            if (result.err) {
+              const { code } = result.err;
```
Comment (NouraldinS) [inline public/js/cohortStudent.js]:
```js
const { code, detail } = result.err;
```

File: src/views/cohort.hbs [hunk 14]
```diff
@@ -0,0 +1,60 @@
+<section class="content">
+  <section>
+    {{#if err}}
+    <h3>{{err}}</h3>
```
Comment (NouraldinS) [inline src/views/cohort.hbs:4]: Fix indentation.

File: src/views/cohortStudents.hbs [hunk 15]
```diff
@@ -0,0 +1,51 @@
+<section class="content">
```
Comment (NouraldinS) [inline src/views/cohortStudents.hbs:1]: Fix indentation.

File: public/css/cohorts.css [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+select.styled {
+  vertical-align: top;
+  text-overflow: ellipsis;
+  display: inline-block;
+
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+
+  -webkit-border-radius: 3px;
```
Comment (ali-7) [inline public/css/cohorts.css]: delete this

File: public/css/cohorts.css [hunk 17]
```diff
@@ -0,0 +1,166 @@
+/* section */
```
Comment (ali-7) [inline public/css/cohorts.css]: remove it

File: public/css/cohorts.css [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+  padding-right: 35px;
+  background: linear-gradient(45deg, transparent 50%, #aaa 50%),
+    linear-gradient(135deg, #aaa 50%, transparent 50%),
+    linear-gradient(to top, #ddd, #fff);
+  background-position: calc(100% - 15px) calc(50%), calc(100% - 10px) calc(50%),
+    100% 0;
+  background-size: 5px 5px, 5px 5px, 30px 100%;
+  background-repeat: no-repeat;
+}
+/* button */
```
Comment (ali-7) [inline public/css/cohorts.css]: remove it

File: public/css/cohorts.css [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+.button {
+  background-color: #4caf50; /* Green */
```
Comment (ali-7) [inline public/css/cohorts.css]: remove comment

File: public/css/cohorts.css [hunk 20]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+.tabel th {
+  padding-top: 12px;
+  padding-bottom: 12px;
+  text-align: left;
+  background-color: #389dbf;
+  color: white;
+}
+/* sectionAddcoh */
```
Comment (ali-7) [inline public/css/cohorts.css]: remove

File: public/css/cohorts.css [hunk 21]
```diff
@@ -0,0 +1,166 @@
...
+.sectionAddcoh{
+    display: none;
+}
+.sectionAddcoh--visible{
+    display: block;
+    margin-bottom: 4rem;
+
```
Comment (ali-7) [inline public/css/cohorts.css]: fix the indentation

File: public/js/cohort.js (hunk 7 above)
Comment (ali-7) [inline public/js/cohort.js]: remove commented line

File: src/database/queries/getCohortStudents.js [hunk 22]
```diff
@@ -0,0 +1,18 @@
+const dbConnection = require('../db_connection');
+
+const getCohortStudentsQuery = (id, cb) => {
+  console.log(id);
```
Comment (ali-7) [inline src/database/queries/getCohortStudents.js]: remove console.log

File: src/database/queries/getCohortStudents.js [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+  
+  console.log(545);
```
Comment (ali-7) [inline src/database/queries/getCohortStudents.js]: remove console.log

File: src/views/cohort.hbs (hunk 14 above)
Comment (ali-7) [inline src/views/cohort.hbs:4]: fix indent (tab)

File: src/views/cohortStudents.hbs [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+  <section class="sectionstd" id="student">
+    <button class="button button2 buttonstd" onclick="addStudents()">ADD Student</button>
+    <hr> {{#if err}}
```
Comment (ali-7) [inline src/views/cohortStudents.hbs]: added to new line

File: src/views/cohortStudents.hbs [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+    <h3>Students of {{cohortname}}</h3>
+    <table class="tabel">
+      <tr>
+        <th>Name</th>
+        <th>Github Link</th>
+        <th>Image</th>
+        <th>Edit</th>
+        <th>Delete</th>
+      </tr>
+      {{#each res}}
```
Comment (ali-7) [inline src/views/cohortStudents.hbs:17]: add indent

**Review by ali-7 — CHANGES_REQUESTED** [review]

File: public/js/cohort.js [hunk 26]
```diff
@@ -0,0 +1,8 @@
+const addStd = document.getElementById('addStd');
+const addCoh = document.getElementById('addCoh');
+
+function addCohort() {
```
Comment (lubnaabd) [inline public/js/cohort.js]: use E6

File: public/js/cohortStudent.js (hunk 10 above)
Comment (lubnaabd) [inline public/js/cohortStudent.js]: use E6

**Review by lubnaabd — CHANGES_REQUESTED** [review]

File: public/css/cohorts.css (hunk 16 above)
Comment (HemaSAli) [reply public/css/cohorts.css]: done

**Review by NouraldinS — CHANGES_REQUESTED** [review]

File: public/js/cohort.js (hunk 26 above)
Comment (HemaSAli) [reply public/js/cohort.js]: really !
should I say that it's your code ?

Comment (NouraldinS) [reply public/js/cohort.js]: @HemaSAli Your pull request :p

File: public/css/cohorts.css (hunk 5 above)
Comment (HemaSAli) [reply public/css/cohorts.css:131]: done Nour , thanks 
and sorry for that ..

File: src/views/cohortStudents.hbs (hunk 15 above)
Comment (HemaSAli) [reply src/views/cohortStudents.hbs:1]: what can I do here ?

File: src/views/cohortStudents.hbs (hunk 25 above)
Comment (HemaSAli) [reply src/views/cohortStudents.hbs:17]: done ,

**Review by lubnaabd — APPROVED** [review]

File: src/views/cohortStudents.hbs (hunk 15 above)
Comment (ali-7) [reply src/views/cohortStudents.hbs:1]: line # 5 start at the beginning.
line # 6 tab 
lines # 8 to 24 need a tab coz it'sunder else
remove space line # 30, 33, 35, 39, 42 
and there's ```</section>```  more than  ```<section>```   line # 51

**Review by ali-7 — APPROVED** [review]

File: src/views/cohort.hbs (hunk 14 above)
Comment (HemaSAli) [reply src/views/cohort.hbs:4]: done .

File: src/views/cohortStudents.hbs (hunk 15 above)
Comment (HemaSAli) [reply src/views/cohortStudents.hbs:1]: thanks Ali , I did it

**Review by NouraldinS — APPROVED** [review]

**Pull request merged**